sql/wsrep_encryption.h

```diff
@@ -0,0 +1,100 @@
+/* Copyright 2019 Codership Oy <info@codership.com>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; version 2 of the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301  USA */
+
+#ifndef WSREP_ENCRYPTION_H
+#define WSREP_ENCRYPTION_H
+
+#include "wsrep/encryption_service.hpp"
+#include <vector>
+
+class Wsrep_encryption_service : public wsrep::encryption_service
+{
+public:
+
+  static Wsrep_encryption_service* instance()
+  {
+    if (m_instance == 0)
+    {
+      m_instance = new Wsrep_encryption_service();
+    }
+    return m_instance;
+  }
+
+  static void destroy()
+  {
+    if (m_instance)
+    { 
+      delete m_instance;
+    }
+  }
+
+  int do_crypt(void**                ctx,
+               wsrep::const_buffer&  key,
+               const char            (*iv)[32],
+               wsrep::const_buffer&  input,
+               void*                 output,
+               bool                  encrypt,
+               bool                  last);
+
+  int set_encryption_key(wsrep::provider& provider, std::vector<unsigned char>& key);
+
+  bool enabled();
+
+  /**
+   * Set encryption key. Serialize it and send to provider.
+   * 
+   * @param[in] key Pointer to encryption key
+   * @param[in] size Length of encryption key
+   * @param[in] version Key version used
+   */
+  bool set_encryption_key(const void* key, size_t size, unsigned int version);
```

Comment:
Why the return value is `bool` instead of int? What do the return values mean? It would be nice to be consistent with other wsrep-lib conventions, use int as return value and have value zero denote success.

Reply:
Aligned to use wsrep-lib conventions